src/backend/distributed/commands/role.c

```diff
@@ -124,6 +124,13 @@ RoleSpecToObjectAddress(RoleSpec *role, bool missing_ok)
 		Oid roleOid = get_rolespec_oid(role, missing_ok);
 		ObjectAddressSet(*address, AuthIdRelationId, roleOid);
 	}
+	else
+	{
+		/*
```

Comment:
I don't think we can reliably do this. What about the callers? We should always carry this edge case with us.

Instead, can we add an exception in `DistOpsHasInvalidObject` , similar to `StatementContainsIfExist` such as `StatementUsesALLRoles` ?

Reply:
I am going to close that PR and resolve the issue at PR #6088.